### PR TITLE
[BTC] New hashtype logic

### LINKED ISSUE #1009: Move BTC script-from-address and hashtype logic to wallet core
Move some BTC transaction related logic to wallet core, so that logic is not in the apps:
- logic to create input scripts (based on UTXO address and coin type)
- logic to select hashType from coin type.

```swift
    static func mapScripts(utxos: [BitcoinUnspentTxOutput]) -> [String: Data] {
        var scriptHashes = [String: Data]()
        // set redeem script for segwit
        for unspent in utxos {
            let address = unspent.address
            let utxo = BitcoinUnspentTransaction(unspent: unspent)
            if unspent.path.purpose != .bip84 {
                continue
            }
            let script = BitcoinScript(data: utxo.script)
            if let scriptHash = script.matchPayToScriptHash() {
                guard let addr = BitcoinAddress(string: address) else {
                    continue
                }
                scriptHashes[scriptHash.hexString] = BitcoinScript.buildPayToWitnessPubkeyHash(hash: addr.keyhash).data
            } else if let scriptHash = script.matchPayToWitnessPublicKeyHash() {
                scriptHashes[scriptHash.hexString] = BitcoinScript.buildPayToPublicKeyHash(hash: scriptHash).data
            }
        }
        return scriptHashes
    }

    var hashType: UInt32 {
        // set fork hash type for BCH
        if coin == .bitcoinCash {
            return (BitcoinSigHashType.all.rawValue | BitcoinSigHashType.fork.rawValue)
        }
        return BitcoinSigHashType.all.rawValue
    }
```

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Move BTC hashtype logic to wallet core.
Fixes #1009
New HashTypeForCoin helper method to TWBitcoinScript, incorporate in signing tests.

## Testing instructions

Standard unit tests, incl. ios and android.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by catenocrypt** [comment]
> let's do separate pr?

I'd rather keep together these 2 related changes, so I can update tests and sample apps at once.

File: samples/osx/cocoapods/Podfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ platform :osx, '10.12'
 
 target 'WalletCoreExample' do
   use_frameworks!
-  pod 'TrustWalletCore', '~> 2.0.0'
+  pod 'TrustWalletCore', '~> 2.1.0'
```
Comment (vikmeup) [inline samples/osx/cocoapods/Podfile:5]:
```suggestion
  pod 'TrustWalletCore'
```

Can we always use latest?

Comment (catenocrypt) [reply samples/osx/cocoapods/Podfile]: +1

File: samples/osx/cocoapods/Podfile.lock [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-  SwiftProtobuf: 2cbd9409689b7df170d82a92a33443c8e3e14a70
-  TrustWalletCore: 43c900920c15fd1ccb8774a051b1055a660f2e91
+  SwiftProtobuf: ecbec1be9036d15655f6b3443a1c4ea693c97932
+  TrustWalletCore: cd0373c69fd9bf92700783287544ce6a9aab041b
 
-PODFILE CHECKSUM: 9f88ed3f7ceadcf38049ea37584a81f309d04e83
+PODFILE CHECKSUM: a34fe0b289ed6fb4db3ae7964f62fa6b21ff0c0d
 
-COCOAPODS: 1.9.0
+COCOAPODS: 1.9.2
```
Comment (hewigovens) [inline samples/osx/cocoapods/Podfile.lock:24]: latest is 1.9.3

**Pull request merged**